src/python/__init__.py.in

```diff
@@ -3,14 +3,18 @@
 
 from onnxruntime_genai import _dll_directory
 
+__version__ = "@VERSION_INFO@"
+__id__ = "@TARGET_NAME@"
+
 try:
-    # Try importing onnxruntime_genai. If an ImportError is raised,
-    # it could be because the cuda dlls could not be found on windows.
-    # Try adding the cuda dlls path to the dll search directory
-    # and import onnxruntime_genai again.
-    from onnxruntime_genai.@PACKAGE_DIR_NAME@ import *
-except ImportError:
-    _dll_directory.add_dll_directory()
     from onnxruntime_genai.@PACKAGE_DIR_NAME@ import *
-
-__version__ = "@VERSION_INFO@"
+except ImportError as e:
+    if __id__ == "onnxruntime-genai-cuda":
+        # Try importing onnxruntime_genai. If an ImportError is raised,
+        # it could be because the cuda dlls could not be found on windows.
+        # Try adding the cuda dlls path to the dll search directory
+        # and import onnxruntime_genai again.
+        _dll_directory.add_dll_directory()
```

Comment:
Thanks for improving the error message. I would have a bit of feature creep on this PR: could you also improve the error message when CUDA is installed but not the expected version?

I just installed CUDA on another machine and the CUDA version is now v12.5, and cudart64_110.dll is now cudart64_12.dll, leading to a generic error message when loading the Python module: `ImportError: DLL load failed while importing onnxruntime_genai: The specified module could not be found.`

![image](https://github.com/microsoft/onnxruntime-genai/assets/8584604/a302cdfc-55b7-4041-8f81-6ba13a848f3b)

Reply:
Thanks @mmaitre314. I think this is a good idea. It will require some more work in figuring out the user installed cuda toolkit version. I will get to this in another PR.